src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -101,6 +101,31 @@ public String getFilteredValidRequestJson(String requestJson) throws IOException
         return null;
     }
 
+    /**
+     * Evaluates request metadata and returns a boolean based on whether the request data
+     * passes all sanity checks.
+     * @throws JsonProcessingException or JsonMappingException
+     */
+    @Override
+    public Boolean isValidRequestMetadataJson(String requestJson)
+            throws JsonMappingException, JsonProcessingException {
+        if (StringUtils.isAllBlank(requestJson)) {
+            return false;
+        }
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> requestJsonMap = mapper.readValue(requestJson, Map.class);
+        boolean isCmoRequest = isCmoRequest(requestJsonMap);
+        boolean hasRequestId = hasRequestId(requestJsonMap);
+
+        // if cmo filter is enabled then skip request if it is non-cmo
+        if (igoCmoRequestFilter && !isCmoRequest) {
+            LOG.info("CMO request filter enabled - skipping non-CMO request: "
+                    + requestJsonMap.get("requestId"));
+            return false;
```

Comment:
[requested change] Similar as above comment. Perhaps we should have a method that returns the request Id extracted from the provided JSON.